src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforceStreamingSource.java

```diff
@@ -0,0 +1,183 @@
+    try {
+      config.validate(); // validate when macros are not substituted
+      config.ensurePushTopicExistAndWithCorrectFields(); // run when macros are not substituted
+
+      String query = config.getPushTopicQuery();
+
+      if (query != null && !query.isEmpty()
+        && !config.containsMacro(SalesforceStreamingSourceConfig.PROPERTY_PUSHTOPIC_QUERY)) {
+
+        Schema schema = SalesforceSchemaUtil.getSchema(config.getAuthenticatorCredentials(),
+                                                       SObjectDescriptor.fromQuery(query));
+        pipelineConfigurer.getStageConfigurer().setOutputSchema(schema);
+      }
+    } catch (ConnectionException e) {
+      throw new RuntimeException("There was issue communicating with Salesforce", e);
+    }
+  }
+
+  @Override
+  public JavaDStream<StructuredRecord> getStream(StreamingContext streamingContext) throws ConnectionException {
+    config.validate(); // validate when macros are substituted
+    config.ensurePushTopicExistAndWithCorrectFields(); // run when macros are substituted
+
+    this.schema = streamingContext.getOutputSchema();
+
+    if (this.schema == null) { // if was not set in configurePipeline due to fields containing macro
+      this.schema = SalesforceSchemaUtil.getSchema(config.getAuthenticatorCredentials(),
+                                                   SObjectDescriptor.fromQuery(config.getPushTopicQuery()));
+    }
+    LOG.debug("Schema is " + schema.toString());
+
+    JavaStreamingContext jssc = streamingContext.getSparkStreamingContext();
+
+    return jssc.receiverStream(new SalesforceReceiver(this.config.getAuthenticatorCredentials(),
+                                                      this.config.getPushTopicName())).
+      map((Function<String, StructuredRecord>) this::getStructuredRecord).filter(Objects::nonNull);
+  }
+
+  private StructuredRecord getStructuredRecord(String jsonMessage) throws ConnectionException {
+    try {
+      StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+
+      LOG.debug("Parsing json " + jsonMessage);
+
+      JSONObject sObjectFields = new JSONObject(jsonMessage) // throws a JSONException if failed to decode
+        .getJSONObject("data") // throws a JSONException if not found
+        .getJSONObject("sobject");
+
+      for (Map.Entry<String, Object> entry : sObjectFields.toMap().entrySet()) {
+        String fieldName = entry.getKey();
+        Object value = entry.getValue();
+
+        Schema.Field field = schema.getField(fieldName, true);
+
+        if (field == null) {
+          continue; // this field is not in schema
+        }
+
+        builder.set(field.getName(), value);
+      }
+      return builder.build();
+    } catch (Exception ex) {
+      switch (config.getErrorHandling()) {
+        case SalesforceStreamingSourceConfig.ERROR_HANDLING_SKIP:
+          LOG.info(String.format("Cannot process json '%s', skipping it.", jsonMessage), ex);
```

Comment:
1. why do you use String.format with logger?
2. are you sure about INFO level, not a WARN?